### ISSUE TITLE
Please make the class matching case insenstive

### ISSUE THREAD

**Issue by ksours** [sebastianbergmann/phpunit#4498]
Recent versions of PHP Unit have added the requirement that the filename for a test be a prefix of the classname for the test.  The current requirement does a case sensitive compare.  My current preference is for mixed case classnames and all lower case filenames (though we do use mytestTest.php to comply with previous phpunit conventions).  It's not the end of the world to have to rename all of the tests, but it's a pain.  If there isn't a compelling reason to have a case sensitive compare, it will break fewer existing test setups to do it case insensitive.

For instance
some/path/mytestTest.php

should allow
```
<?php
class some_path_myTestTest extends PHPUnit\Framework\TestCase
{
    public function testTrue(): void
    {
        $this->assertTrue(true);
    }
}
```

**Comment by sebastianbergmann** [comment]
@flow-control Do you have an idea what this is about? Thanks!

**Comment by flow-control** [comment]
Hey @ksours,

seems like there is some content missing at the end of your issue report. I still played around with it anyway and found that this case sensitive behaviour you mention occurs when
- you call PHPUnit with a file name as in `phpunit some/path/mytestTest.php`
- **and** the test class in this file is in a namespace
- **and** the class name does not case sensitive match for `mytestTest`

When you remove the namespace or rename the test class to match (case sensitive) `mytestTest` everything works as expected.

The issue lies in [these lines](https://github.com/sebastianbergmann/phpunit/blob/master/src/Runner/StandardTestSuiteLoader.php#L58-L59) in the `StandardTestSuiteLoader`:

``` php
if (substr($loadedClass, $offset) === $suiteClassName &&
    basename(str_replace('\\', '/', $loadedClass)) === $suiteClassName) {
```

This class has been deprecated along with some other classes and methods to clean things up. Nevertheless you might have a point here as the behaviour is inconsistent: enforcing case sensitive if class is in namespace and allowing default PHP case insensitive class name in case the class is not in a namespace.

@sebastianbergmann: I could relatively fast create a fix for this and allow for case insensitive class names also in case the class is in a namespace, this would also align behaviour with PHP, where class names are handled case insensitive. What do you think?

**Comment by flow-control** [comment]
This works:

```php
<?php declare(strict_types=1);
namespace PHPUnit\TestFixture;

use PHPUnit\Framework\TestCase;

class TestWithTest extends TestCase
{
    public function testTrue(): void
    {
        $this->assertTrue(true);
    }
}
```

```bash
./phpunit tests/_files/TestWithTest.php 
PHPUnit 9.5-g09ba0b8d9 by Sebastian Bergmann and contributors.

Runtime:       PHP 7.4.11
Configuration: /home/bowman/Work/flow-control/phpunit/phpunit.xml

.                                                                   1 / 1 (100%)

Time: 00:00.005, Memory: 4.00 MB
```

this does **NOT** work

```php
<?php declare(strict_types=1);
namespace PHPUnit\TestFixture;

use PHPUnit\Framework\TestCase;

class testwithtest extends TestCase
{
    public function testTrue(): void
    {
        $this->assertTrue(true);
    }
}
```

```bash
./phpunit tests/_files/TestWithTest.php 
Class 'TestWithTest' could not be found in '[..]/tests/_files/TestWithTest.php'.
```

**Comment by sebastianbergmann** [comment]
>this would also align behaviour with PHP, where class names are handled case insensitive. What do you think?

Makes sense to me, thanks.

**Comment by ksours** [comment]
I missed a code tag in my original report.  Updated it so you should be seeing it.  I has been inadvertently looking at the PHPUnit 9.0.0 code when I made the report and it appears to have changed in 9.4.2.

Sorry about that.

**Comment by flow-control** [comment]
Hey @ksours,

no problem. When the class is named `some_path_mytestTest` this has nothing to do with case sensitive Vs case insensitive, but the path should not be in the class name at all.
Could you check if the pull request I created fixes the behaviour for you and if not please provide a minimum example on how to reproduce the error.

Kind regards
Florian

**Comment by flow-control** [comment]
@ksours the fix to this has been released with [`9.4.3`](https://github.com/sebastianbergmann/phpunit/blob/9.4.3/ChangeLog-9.4.md#943---2020-11-10)
